elastic/apm/index-template.json

```diff
@@ -1,9 +1,28 @@
 {
-    "index_patterns": ["traces-apm*", "logs-apm*", "metrics-apm*"],
-    "data_stream": {},
-    "_meta": {
-        "description": "Placeholder template that'd be used when the x-pack-apm car is not used, e.g. in test mode"
-    },
-    "composed_of": ["apm@settings"],
-    "priority": 1
+  "index_patterns": ["traces-apm*", "logs-apm*", "metrics-apm*"],
+  "data_stream": {},
+  "_meta": {
+      "description": "Placeholder template that'd be used when the x-pack-apm car is not used, e.g. in test mode"
+  },
+  "composed_of": ["traces@custom", "metrics@custom", "logs@custom"],
+  "priority": 101,
+  {# in test-mode, following field is not handled by ingest pipeline so it returns a mapping error. #}
+  {# preventing it by explicitly set the mapping #}
+  "template": {
+    "mappings": {
+      "properties": {
+        "metricset": {
+          "properties": {
+            "samples": {
+              "properties": {
+                "value": {
+                  "type": "long"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
```

Comment:
@axw I've added this to fix the tests which is there to check the track very quickly for [syntax errors](https://esrally.readthedocs.io/en/stable/adding_tracks.html?highlight=test%20mode#adding-support-for-test-mode). tests were falling due to mapping errors, which shouldn't happen when the proper ingest pipeline/template is installed. In test mode it is not enabled by default so this is a workaround until we enable the apm plugin by default in ES.